src/apps/polymer/vIBCEscrow.sol

```diff
@@ -0,0 +1,181 @@
+ * and contracts. As a result, if we trust a channel, we also imply that we trust
+ * the contract on the other end of that channel. This is unlike "traditional" chain
+ * mappings where there may be may addresses on the other end.
+ * As a result, we are allowed to just append our address to the package and then trust that.
+ * Because if someone trust the channel (which is a requirement) then they must also
+ * trust the account AND the set value.
+ */
+contract IncentivizedPolymerEscrow is APolymerEscrow, IbcReceiverBase, IbcReceiver {
+    error ChannelNotFound();
+    error UnsupportedVersion();
+
+    uint constant POLYMER_SENDER_IDENTIFIER_START = 0;
+    uint constant POLYMER_SENDER_IDENTIFIER_END = 32;
+    uint constant POLYMER_PACKAGE_PAYLOAD_START = 32;
+
+    bytes32[] public connectedChannels;
+    string constant VERSION = 'vIBC Escrow 1.0';
+
+    // Make a shortcut to save a bit of gas.
+    bytes32 immutable ADDRESS_THIS = bytes32(uint256(uint160(address(this))));
+
+    constructor(address sendLostGasTo, address dispatcher)
+        APolymerEscrow(sendLostGasTo)
+        IbcReceiverBase(IbcDispatcher(dispatcher))
+    {}
+
+    //--- IBC Channel Callbacks ---//
+
+    function onOpenIbcChannel(
+        string calldata version,
+        ChannelOrder /*  */,
+        bool,
+        string[] calldata,
+        CounterParty calldata counterparty
+    ) external view onlyIbcDispatcher returns (string memory selectedVersion) {
+        if (counterparty.channelId == bytes32(0)) {
+            // ChanOpenInit
+            if (
+                keccak256(abi.encodePacked(version)) != keccak256(abi.encodePacked(VERSION))
+            ) revert UnsupportedVersion();
+        } else {
+            // ChanOpenTry
+            if (
+                keccak256(abi.encodePacked(counterparty.version)) != keccak256(abi.encodePacked(VERSION))
+            ) revert UnsupportedVersion();
+        }
+        return VERSION;
+    }
+
+    function onConnectIbcChannel(
+        bytes32 channelId,
+        bytes32,
+        string calldata counterpartyVersion
+    ) external onlyIbcDispatcher {
+        if (
+            keccak256(abi.encodePacked(counterpartyVersion)) != keccak256(abi.encodePacked(VERSION))
+        ) revert UnsupportedVersion();
+        connectedChannels.push(channelId);
+    }
+
+    function onCloseIbcChannel(bytes32 channelId, string calldata, bytes32) external onlyIbcDispatcher {
+        unchecked {
+        // logic to determin if the channel should be closed
+        bool channelFound = false;
+        for (uint256 i = 0; i < connectedChannels.length; ++i) {
+            if (connectedChannels[i] == channelId) {
+                delete connectedChannels[i];
+                channelFound = true;
+                return;
+                // We could also break but early return saves gas.
+            }
+        }
+        if (!channelFound) revert ChannelNotFound();
+
+        }
+    }
```

Comment:
Note: `closeIbcChannel` is not implemented yet. 
Once it's supported, you also need to a `triggerChannelClose` func to trigger the call to Dispatcher's `closeIbcChannel`, similar to our [example Mars contract](https://github.com/open-ibc/vibc-core-smart-contracts/blob/7777f17042a35c41eb597cf61c9676e1a0ae70fc/contracts/examples/Mars.sol#L54-L55)

Reply:
It is fine. I don't see any reason to want to close the channel. I copied this from the universal channels contract. Instead, should we just always revert?